scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -2185,17 +2205,28 @@ private void sendMessageWithSession(final RawData message, final Connection conn
 				}
 				message.onConnecting();
 				Handshaker previousHandshaker = connection.getOngoingHandshake();
-				SessionTicket ticket;
-				SessionId sessionId;
+				SessionTicket ticket = null;
+				SessionId sessionId = null;
 				if (session != null) {
 					sessionId = session.getSessionIdentifier();
 					ticket = session.getSessionTicket();
-					connectionStore.removeFromEstablishedSessions(session, connection);
+					if (!probing) {
+						connectionStore.removeFromEstablishedSessions(session, connection);
+					}
 				} else {
-					sessionId = connection.getSessionIdentity();
-					ticket = connection.getSessionTicket();
+					if (!full) {
+						sessionId = connection.getSessionIdentity();
+						ticket = connection.getSessionTicket();
+					}
+					probing = false;
+				}
+				if (probing) {
+					// Only reset the resumption trigger, but keep the session for now
+					// the session will be reseted with the first received data
+					connection.setResumptionRequired(false);
```

Comment:
I didn't get this part.

Reply:
Maybe you can give some more hints, what is hard to get?

The intention of probing is documented in the javadoc of `ResumingClientHandshaker`.

```
if (probing) {
	// Only reset the resumption trigger, but keep the session for now
	// the session will be reseted with the first received data
	connection.setResumptionRequired(false);
} else {
	connection.resetSession();
}
```

And in `processRecord` ("the session will be reseted with the first received data"):

```
if (handshaker != null && handshaker.isProbing()) {
	// received record, probe successful
	if (connection.hasEstablishedSession()) {
		connectionStore.removeFromEstablishedSessions(connection.getEstablishedSession(), connection);
	}
	connection.resetSession();
	handshaker.resetProbing();
	LOGGER.debug("handshake probe successful {}", connection.getPeerAddress());
}
```

Reply:
I try to be clearer. 

I saw the documentation in `ResumingClientHandshaker` :+1: 

What I understand is that with probing mode, we try to do a handshake and if it failed we didn't remove the original/previous connection. (because maybe we lost connectivity but not connection)

What I didn't get is the line : `connection.setResumptionRequired(false);`
I knew there is a comment just above, but I didn't get why using probing means "no resumption required"

Reply:
Ah OK, no it requires one resumption, but this one is started in the lines afterwards.
Reset that mark, either by `connection.setResumptionRequired(false)` or 
`connection.resetSession()`, suppresses further resumptions without new trigger.
(see javadoc of `connection.resetSession()`).

Reply:
Ok I finally get it.
We will initiate a handshake and so in case where resumption was required, we currently take it into account and so we go back in a "normal state"
Generally we go back to this normal state/"no resumption required" using resetSession()
But here we want to keep the session and so we just note that resumption is taking into account.